core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -839,7 +851,26 @@ private[spark] class MapOutputTrackerMaster(
   }
 
   def registerMapOutput(shuffleId: Int, mapIndex: Int, status: MapStatus): Unit = {
-    shuffleStatuses(shuffleId).addMapOutput(mapIndex, status)
+    if (enableOptimizeCompressedMapStatus && status != null) {
```

Comment:
Can `MapStatusesSerDeserBenchmark` be used to demonstrate performance improvements?

Reply:
+1 to @LuciferYang's comment above.
Also, please add more details to the PR description on what is the optimization/how it helps,  etc.